### PR TITLE
GG-19218 Invalidate from cache old topology snapshots.

### PR DESCRIPTION
**Pull request by nva**

### CODE REVIEW

File: modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/web/socket/WebSocketsManager.java [hunk 1]
```diff
@@ -55,9 +61,16 @@
  * Web sockets manager.
  */
 @Service
+@EnableScheduling
 public class WebSocketsManager {
     /** */
-    private static final Logger log = LoggerFactory.getLogger(WebSocketsManager.class);
+    private static final IgniteLogger log = new Slf4jLogger(LoggerFactory.getLogger(WebSocketsManager.class));
```
Comment (akuznetsov-gridgain) [inline modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/web/socket/WebSocketsManager.java]: Why do we need IgniteLogger  ?

File: modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/web/socket/WebSocketsManager.java [hunk 2]
```diff
@@ -80,11 +93,13 @@
...
      * Default constructor.
      */
-    public WebSocketsManager() {
+    public WebSocketsManager(Ignite ignite) {
         agents = new ConcurrentLinkedHashMap<>();
         browsers = new ConcurrentHashMap<>();
         clusters = new ConcurrentHashMap<>();
         requests = new ConcurrentHashMap<>();
+
+        new IgniteThread(new ClusterTimeoutWorker(ignite.name(), "session-timeout-worker", log)).start();
```
Comment (akuznetsov-gridgain) [inline modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/web/socket/WebSocketsManager.java]: I do not like this. How about to use Spring "@Scheduled" (or something like this) instead?

File: modules/web-console/web-console-server/pom.xml [hunk 3]
```diff
@@ -136,6 +136,12 @@
             <version>${web-console.ignite.version}</version>
         </dependency>
 
+        <dependency>
```
Comment (akuznetsov-gridgain) [inline modules/web-console/web-console-server/pom.xml]: Do we need this dependency now?

File: modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/Application.java [hunk 4]
```diff
@@ -24,7 +24,6 @@
  * Web console launcher.
  */
 @SpringBootApplication
-@EnableScheduling
```
Comment (akuznetsov-gridgain) [inline modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/Application.java]: Not sure if it is good idea to move this annotation to service.
We may need scheduling in other places later.
I would keep it here.

File: modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/web/socket/WebSocketsManager.java [hunk 5]
```diff
@@ -55,10 +56,14 @@
  * Web sockets manager.
  */
 @Service
+@EnableScheduling
```
Comment (akuznetsov-gridgain) [inline modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/web/socket/WebSocketsManager.java]: I think it is better to have this annotation on Application.

File: modules/web-console/web-console-server/pom.xml (hunk 3 above)
Comment (nva) [reply modules/web-console/web-console-server/pom.xml]: no

File: modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/web/socket/WebSocketsManager.java (hunk 5 above)
Comment (nva) [reply modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/web/socket/WebSocketsManager.java]: Application is entry point. If you create another module that depends this - scheduling will work only after you add this anotation to new Application

File: modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/Application.java (hunk 4 above)
Comment (nva) [reply modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/Application.java]: I think will be better to create separate configuration.

**Pull request merged**